### ISSUE TITLE
Large Amounts of Addon cars breaks shop

### ISSUE THREAD

**Issue by mmaust005** [HumanTree92/esx_advancedvehicleshop#60]
**Describe the Bug | A Clear & Concise Description of Bug**

Had no issues adding addon vehicles to shops. the issue I am having is after adding the Vanilla Works Extended pack along with custom police cars (redneck, and a few others) the shops break. I am hosting on a remote server which is an identical clone to my local test server. I tested a few cars from the Vanilla Extended works and Rednecks car packs. I can get them to appear in the shops with the correct names. I also modified the categories databases to reflect better categories, ie removed the trucks from vans and put them in a Truck category for the car shop. Everything worked fine during testing so I figured I could add the whole pack(s). I made the edits in policejob as the readme stated, I applied the esx_lscustom fix, Like I said adding vehicles in small batches works fine. Currently, at the car shop, you teleport to the InsideShop location but the menu disappears and you cannot see the character. The police dealership clips you just under the ground but same effects as the car dealership. Using F8 or the server console through txAdmin to restart the resource works to get the character unstuck as usual. I was unsure if it was a resource loading time issue so I went through with vMenu and spawned all the vehicles to make sure I already had them cashed but no difference.

Not sure if there is a limitation with the script or not. With the policejob I have 4 departments and with needing a separate line in the database there are about 500+/- entries for the police vehicles. Vanilla Works Extended added about 319+/- vehicles and other things. Does the policejob config still need to be adjust adjusted to show "authorized" vehicles for the given rank for them to appear? It all worked fine but I know the additional vehicles are most likely the cause but would like to find a solution to still be able to use them.

The only things I changed were in the config to move the locations out into the open to see if the vehicles spawning too high up and not counting the character within the zones.

**Questions**
Have you made changes to anything in `client/main.lua` or `server/main.lua`?: No
Have you looked through the Closed Topics?: Yes.
Have you looked through the Wiki?: Yes
Are you using a Pre-Installed ESX? If Yes who?: No
Are you using es_extended? If Yes what Version? (Example: 1.2):  v1 Final
Are you using Essentialmode?: Yes
Are you using ExtendedMode?: No
Are you using OneSync?: Yes, Infinity
Linux or Windows?:  Windows

**Comment by mmaust005** [comment]
UPDATE: I have seen an error in the client console that does not appear in the server console 

This appears when trying to access the Vehicle Shop:
`SCRIPT ERROR: @esx_advancedvehicleshop/client/main.lua:1301: bad argument #1 to "insert" (table expected, got nil)`

This appears when trying to access the Police Shop:
`SCRIPT ERROR: @esx_advancedvehicleshop/client/main.lua:473: bad argument #1 to "insert" (table expected, got nil)`

These also come up at the Vehicle Shop:

`[esx_advancedvehicleshop] [ERROR] Boat "asbo" does not exist
[esx_advancedvehicleshop] [ERROR] Boat "brioso2" does not exist
[esx_advancedvehicleshop] [ERROR] Boat "caracara2" does not exist
[esx_advancedvehicleshop] [ERROR] Boat "club" does not exist
[esx_advancedvehicleshop] [ERROR] Boat "coquette4" does not exist
[esx_advancedvehicleshop] [ERROR] Boat "asbodrafter" does not exist
[esx_advancedvehicleshop] [ERROR] Boat "dukes3" does not exist
[esx_advancedvehicleshop] [ERROR] Boat "dynasty" does not exist
[esx_advancedvehicleshop] [ERROR] Boat "emerus" does not exist
[esx_advancedvehicleshop] [ERROR] Boat "everon" does not exist
[esx_advancedvehicleshop] [ERROR] Boat "formula" does not exist
[esx_advancedvehicleshop] [ERROR] Boat "formula2" does not exist
[esx_advancedvehicleshop] [ERROR] Boat "furia" does not exist` 

etc, etc. 59 total. A mixture of Vallila Works Extended and Base Game cars.

**Comment by HumanTree92** [comment]
Normally when you see `[esx_advancedvehicleshop] [ERROR] Boat "VEHICLE_MODEL" does not exist` you need to remove that vehicle from the shop b/c the script can't find said vehicles. Most of those vehicles are DLC vehicles that FiveM themselves have not added and you must add yourself.

I am working on a complete re-write of this script which will have the DLC vehicles not normally in the shop.
There is no Limitations on this script however there are some Limitations with esx_menu_default that i can't control.

Few questions:
1. How many Categories do you have for `vs_police_categories`
2. You say that you have 4 Depts for the Police job. Are you saying you have 4 different Police jobs all using the default police job or are you saying that you have 4 different police jobs where each one is using something like police, sheriff, etc as the job name.

**Comment by mmaust005** [comment]
The departments are structured as follows  LSPD Recruit, LSPD Officer, LSPD Sergeant, LSPD Lieutenant, LSPD Undercover, and LSPD Detective. Names are the same as ranks (lspdrecruit, lspdofficer, etc, etc.) but 2 Total. I've added them to a single policejob config for weapon authorizations and other things. It works fine for the most part. If it's causing a conflict with the vehicle shop I can try and find a way to condense them down. I'm looking at a decently consistent server population and was only adding the specific ranks to allow certain departments access to department-only vehicles for less admin management. If need be I can always do it the old fashion way and manually give the person the vehicle if I have two. The Vanilla Extended Works the main thing I would like to get working. 

Would it be better to run a separate policejob resource for each department? Or bundle them into one? Are they hardcoded to only have just grades of 0, 1, 2, 3, or something?

I'll go through the database I exported and remove the DLC/Missing cars and clean that up to see if it helps as well. 

Thank you for your time.

**Comment by mmaust005** [comment]
Well, I feel like an idiot. After sleeping on it and reviewing the errors I figured it out. Was 1 line in the SQL file I made, there was a misspelling in one of the categories columns that were throwing the script error for each shop. Works perfectly fine. I thank you for your time and I appreciate it. Your work is amazing and I look forward to the next release like you mentioned above.


![20210608143315_1](https://user-images.githubusercontent.com/47438478/121239231-f7877080-c866-11eb-9c7e-5285cbebd15b.jpg)
![20210608143308_1](https://user-images.githubusercontent.com/47438478/121239236-fa826100-c866-11eb-8a87-994e2310657a.jpg)
![20210608143303_1](https://user-images.githubusercontent.com/47438478/121239245-fce4bb00-c866-11eb-8358-d17c0d2f98e8.jpg)
![20210608143231_1](https://user-images.githubusercontent.com/47438478/121239248-feae7e80-c866-11eb-95a3-034210adaf85.jpg)
